src/Raven.Server/Web/System/DatabasesHandler.cs

```diff
@@ -83,6 +87,74 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
             }
         }
 
+        [RavenAction("/topology/modify", "POST", AuthorizationStatus.Operator)]
+        public async Task ModifyTopology()
+        {
+            var dbName = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name").Trim();
+            var raftRequestId = GetRaftRequestIdFromQuery();
+
+            await ServerStore.EnsureNotPassiveAsync();
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var json = await context.ReadForDiskAsync(RequestBodyStream(), "Database Topology");
+                var databaseTopology = JsonDeserializationCluster.DatabaseTopology(json);
+
+                if (LoggingSource.AuditLog.IsInfoEnabled)
+                {
+                    var clientCert = GetCurrentCertificate();
+
+                    var auditLog = LoggingSource.AuditLog.GetLogger("DbMgmt", "Audit");
+                    auditLog.Info($"Database \'{dbName}\' record has been modified by {clientCert?.Subject} ({clientCert?.Thumbprint})");
+                }
+
+                // Validate Database Name
+                DatabaseRecord databaseRecord;
+                using (context.OpenReadTransaction())
+                    databaseRecord = ServerStore.Cluster.ReadDatabase(context, dbName, out var index);
+
+                if (databaseRecord == null)
+                {
+                    throw new DatabaseDoesNotExistException($"Database {dbName} Record not found when trying to modify database topology");
+                }
+
+                // Validate Topology
+                var clusterTopology = ServerStore.GetClusterTopology(context);
+                var databaseAllNodes = databaseTopology.AllNodes;
+                foreach (var node in databaseAllNodes)
+                {
+                    if (clusterTopology.Contains(node) == false)
+                        throw new ArgumentException($"Failed to modify database {dbName} topology, because we don't have node {node} (which is in the new topology) in the cluster.");
+
+                    if (databaseRecord.Topology.RelevantFor(node) == false)
+                    {
+                        ValidateNodeForAddingToDb(dbName, node, databaseRecord, clusterTopology, baseMessage: $"Can't modify database {dbName} topology");
+                    }
+                }
+                databaseTopology.ReplicationFactor = Math.Min(databaseTopology.Count, clusterTopology.AllNodes.Count);
+
+                // Update Topology
+                var update = new UpdateTopologyCommand(dbName, SystemTime.UtcNow, raftRequestId)
+                {
+                    Topology = databaseTopology
+                };
+
+                var (newIndex, _) = await ServerStore.SendToLeaderAsync(update);
+
+                await ServerStore.Cluster.WaitForIndexNotification(newIndex);
```

Comment:
we are waiting here for cluster, if database is loaded, should we wait for it as well? cc @karmeli87

Reply:
remove ```await ServerStore.Cluster.WaitForIndexNotification(newIndex);``` and wait for the raft index in the tests instead?

Reply:
no, using databasesLandlord you can checkk if database is loaded and then get it from there, afterwards you can wait for the command to be applied

Reply:
We don't wait for it when we adding a new node, so I would argue that we don't need to wait for it here as well